metrics-jetty9/src/main/java/io/dropwizard/metrics5/jetty9/InstrumentedConnectionFactory.java

```diff
@@ -1,21 +1,36 @@
 package io.dropwizard.metrics5.jetty9;
 
-import io.dropwizard.metrics5.Timer;
+import java.util.List;
+import java.util.Optional;
+
 import org.eclipse.jetty.io.Connection;
 import org.eclipse.jetty.io.EndPoint;
 import org.eclipse.jetty.server.ConnectionFactory;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.util.component.ContainerLifeCycle;
 
-import java.util.List;
+import io.dropwizard.metrics5.Counter;
+import io.dropwizard.metrics5.Timer;
 
 public class InstrumentedConnectionFactory extends ContainerLifeCycle implements ConnectionFactory {
     private final ConnectionFactory connectionFactory;
     private final Timer timer;
+    private final Optional<Counter> counterMaybe;
 
     public InstrumentedConnectionFactory(ConnectionFactory connectionFactory, Timer timer) {
+        this(connectionFactory, timer, Optional.empty());
+    }
+
+    public InstrumentedConnectionFactory(ConnectionFactory connectionFactory, Timer timer, Counter counter) {
+        this(connectionFactory, timer, Optional.of(counter));
+    }
+
+    private InstrumentedConnectionFactory(ConnectionFactory connectionFactory,
```

Comment:
If we removed `Optional`, I believe the private constructor would be unnecessary.